service/controller/internal/cloudconfig/tccpn.go

```diff
@@ -77,23 +96,38 @@ func (t *TCCPN) Render(ctx context.Context, cr infrastructurev1alpha2.AWSCluster
 		kubeletExtraArgs = append(kubeletExtraArgs, t.config.KubeletExtraArgs...)
 	}
 
+	masterID := 0 // for now we have only 1 master, TODO get this value via render function as argument
```

Comment:
We may need to break up the rendering now anyway and should consider having a render config struct. These 16 different parameters for the function call are not really manageable.